Re-written Getting started page (#759)
MewsSystems/gitbook-connector-api: getting-started/README.md

# Getting started

This page walks you through the key steps to begin developing your integration with the __Mews Connector API__. Whether you're exploring the demo environment, preparing for certification, or validating your first request, this guide covers the tools, environments and concepts you'll need.

> **Ready to try a test call?** Jump to [Step 2: Make your first API call](#step-2-make-your-first-api-call).

## What you’ll need

Before making your first request, make sure you're familiar with the following:

- **Authentication tokens**<br>
  Every API request requires:
  - `ClientToken`: Identifies your application. Issued by Mews.
  - `AccessToken`: Identifies the enterprise or property you're connecting to. Issued by the property.

>   - `Client`: A short string naming your application (used for logging/debugging).

Reviewer: ```suggestion
  - `Client`: A short string naming your application.
```
I would omit this, partners should not be aware how we use this field.

- **Mews environments**<br>

>   Mews provides two primary environments:

Reviewer: ```suggestion
  Mews provides two environments:
```
stating "primary" brings questions like... is there more?

## Step 1: Set up your environment

To begin testing:

1. **Choose a demo property** from the list in [Demo environments](../guidelines/environments.md#demo-environments).
   - Use either the Net pricing or Gross pricing variant depending on your target markets.
2. **Note the tokens** for that property:
   - `ClientToken`
   - `AccessToken`
3. **(Optional)** Log in to [Mews Operations](https://app.mews-demo.com) using the provided email and password to explore the UI and sample data.

You’ll use these credentials to make authenticated API requests.

## Step 2: Make your first API call

Let’s start by calling [Get configuration](../operations/configuration.md#get-configuration), which returns details about the test property — a useful check that your setup is working.

- **Endpoint URL**

```
https://api.mews-demo.com/api/connector/v1/configuration/get
```

- **Request body**
```json
{
  "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
  "AccessToken": "7059D2C25BF64EA681ACAB3A00B859CC-D91BFF2B1E3047A3E0DEC1D57BE1382",
  "Client": "NameOfYourCompanyOrApplication"
}
```

- **Expected response**  

> If successful, the API will return property details:

Author: ```suggestion
If successful, the API will return HTTP status code `200` along with property details in the message body:
```

> If successful, the API will return property details:

Reviewer: idea: what about to say it's return 200 response code with property details (without putting the response here). It's more simple imho.

```json
{
  "NowUtc": "2021-05-05T11:39:29Z",
  "Enterprise": {
    "Id": "c65ea6e9-2340-42f4-9136-ab3a00b6da22",
    "Name": "Connector API Hotel (Net Pricing) TEST",
    "TimeZoneIdentifier": "America/New_York",
    "LegalEnvironmentCode": "US-DC",
    "AccommodationEnvironmentCode": "US",
    ...
  },
  "PaymentCardStorage": {
    "PublicKey": "1100016827"
  }
}
```

> **Requests and responses**: For more on request formatting and response codes, see [Requests](../guidelines/requests.md) and [Responses](../guidelines/responses.md) respectively.

> ## Step 3: Learn the API essentials

Reviewer: I would add mention of best practices here, important page

### Time and data serialization
- All dates and times are in UTC.
- Dates, durations and other data types follow specific [serialization rules](../guidelines/serialization.md).

### Pagination
- Some operations (like [Get all reservations](../operations/reservations.md#get-all-reservations-2023-06-06)) use [pagination](../guidelines/pagination.md) to handle large result sets.

### Rate limits

> - Shared demo credentials may return `429` errors if multiple developers are testing the same property.

Reviewer: ```suggestion
- Shared demo credentials may return `429 - Too many requests` response code if multiple developers are testing the same property.
```
nitpick: to me reading it, it felt like it's gonna return 429 errors, like count of errors 😅
